website/project/model.py

```diff
@@ -3032,3 +3031,72 @@ def approve_embargo(self, user, token):
                 },
                 auth=Auth(user),
             )
+
+class DraftRegistration(AddonModelMixin, StoredObject):
+
+    is_draft_registration = True
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    datetime_initiated = fields.DateTimeField(auto_now_add=True)
+    datetime_updated = fields.DateTimeField(auto_now=True)
+
+    branched_from = fields.ForeignField('node')
+
+    initiator = fields.ForeignField('user')
+
+    registration_metadata = fields.DictionaryField({})
+    registration_schema = fields.ForeignField('metaschema')
+    registered_node = fields.ForeignField('node')
+
+    is_pending_review = fields.BooleanField(default=False)
+
+    schema_name = fields.StringField()
+
+    storage = fields.ForeignField('osfstoragenodesettings')
+
+    # proxy fields from branched_from Node
+    def __getattr__(self, attr):
+        try:
+            return self.__dict__[attr]
+        except KeyError:
+            return getattr(self.branched_from, attr, None)
+
+    def get_comments(self):
```

Comment:
@barbour-em, my gut says this should return a dictionary like:
```python
{
  [QUESTION_ID]: {
    'question': [THE_ORIGINAL_QUESTION],
    'comments': [ORDERED_LIST_OF_COMMENTS]
  }
}
```

We might consider adding a helper or intermediate function that gets a flat list of comments as well.